shared/js/simslot_manager.js

```diff
@@ -46,11 +46,26 @@
         return;
       }
 
+      var allSIMCardDetected = true;
       this._conns.forEach(function iterator(conn, index) {
-        this._instances.push(new SIMSlot(conn, index,
-                             IccManager.getIccById(conn.iccId)));
+        var slot = new SIMSlot(conn, index,
+                               IccManager.getIccById(conn.iccId));
+        this._instances.push(slot);
+        if (slot.isAbsent()) {
+          allSIMCardDetected = false;
+        }
       }, this);
 
+      if (allSIMCardDetected) {
+        this.publishSIMSlotIsReady();
+        return;
```

Comment:
Why do you need to return? We still need to listen to iccdetected event, don't we?